### PR TITLE
Add closing modular window

### LINKED ISSUE #20: (SP: 1) Closing confirmation pop-up window for the Stepper
## Description
Implement the closing confirmation pop-up window as it was implemented for **SignUp**. 

## Design 

Image

## Acceptance Criteria
- [ ] If a user clicks on the close button in the Stepper the pop-up window must appear.
- [ ] The pop-up window must not appear if the user clicks outside the Stepper.
- [ ] The pop-up window must disappear if user click on the area of the Stepper.

### LINKED ISSUE #43: (SP: 1) Confirmation pop-up window.
UserStory: #55 

## Description
Implement the confirmation pop-up window according to the mockup.

## Design 

image

## Acceptance Criteria
- [ ] If a user clicks on the close button in the sign-up the pop-up window must appear.
- [ ] The pop-up window must not appear if the user clicks outside the sign-up form.
- [ ] The pop-up window must disappear if user click on the area of the sign-up form

### PR DESCRIPTION
**Pull request by geersann**
###Issue
Open https://github.com/WebUI-Project-Based-10-nr/Client/issues/20 Implement closing modular window
Open https://github.com/WebUI-Project-Based-10-nr/Client/issues/43 Implement closing modular window

###Description
Implement ConfirmDialog in PopupDialog, to warn about unsaved data

###Assignees
assignees: @geersann

### CODE REVIEW

**Review by hanus-kostiantyn — APPROVED** [review]

**Review by dimaP2341 — APPROVED** [review]

File: src/components/popup-dialog/PopupDialog.tsx [hunk 1]
```diff
@@ -22,33 +24,88 @@ const PopupDialog: FC<PopupDialogProps> = ({
...
+          </Typography>
+          <IconButton onClick={cancelCloseHandler} sx={styles.icon}>
+            <CloseIcon />
+          </IconButton>
+        </Box>
+        <Typography
+          id='confirm-dialog-description'
+          sx={styles.dialogDescription}
+        >
+          Are you certain you want to close? Any unsaved changes will be lost
```
Comment (petro97) [inline src/components/popup-dialog/PopupDialog.tsx]: I suppose we should localize this text like {t('..')}

File: src/components/popup-dialog/PopupDialog.tsx [hunk 2]
```diff
@@ -22,33 +24,88 @@ const PopupDialog: FC<PopupDialogProps> = ({
...
+        </Typography>
+        <Box sx={styles.buttonContainer}>
+          <Button
+            color='primary'
+            onClick={confirmCloseHandler}
+            sx={styles.yesButton}
+            variant='contained'
+          >
+            Yes
```
Comment (petro97) [inline src/components/popup-dialog/PopupDialog.tsx]: Please, localize this text as well. There are already added key "yes" in the common.json file.

File: src/components/popup-dialog/PopupDialog.tsx [hunk 3]
```diff
@@ -22,33 +24,88 @@ const PopupDialog: FC<PopupDialogProps> = ({
...
+          </Button>
+          <Button
+            color='secondary'
+            onClick={cancelCloseHandler}
+            sx={styles.noButton}
+            variant='outlined'
+          >
+            No
```
Comment (petro97) [inline src/components/popup-dialog/PopupDialog.tsx]: Please, localize this text as well. There are already added key "no" in the common.json file

File: src/components/popup-dialog/PopupDialog.styles.ts [hunk 4]
```diff
@@ -12,5 +12,43 @@ export const styles = {
...
     right: { xs: '8px', sm: '20px' },
     top: { xs: '8px', sm: '20px' }
+  },
+  dialogPaper: {
+    padding: '20px',
+    borderRadius: '8px'
+  },
+  dialogTitle: {
+    fontWeight: 'bold',
+    color: '#566970'
```
Comment (petro97) [inline src/components/popup-dialog/PopupDialog.styles.ts]: Regarding the color hex in the css files. We shouldn't create hex each time and duplicate it. The best practice is use variables that are already set in src/styles/app-theme/app.pallete.ts. file. If you took this hex from mockups and in the variables not this color yet please add this hex color to the relevant basic, error, warning,... section.

File: src/components/popup-dialog/PopupDialog.tsx (hunk 1 above)
Comment (geersann) [reply src/components/popup-dialog/PopupDialog.tsx]: This is not necessary, there is a set of questions in json

File: src/components/popup-dialog/PopupDialog.styles.ts (hunk 4 above)
Comment (geersann) [reply src/components/popup-dialog/PopupDialog.styles.ts]: I second, fixed it

**Review by petro97 — APPROVED** [review]
Looks good to me!

File: src/components/popup-dialog/PopupDialog.tsx [hunk 5]
```diff
@@ -22,33 +27,88 @@ const PopupDialog: FC<PopupDialogProps> = ({
...
+          onMouseOver={handleMouseOver}
+          sx={styles.box}
+        >
+          <IconButton onClick={handleClose} sx={styles.icon}>
+            <CloseIcon />
+          </IconButton>
+          <Box sx={styles.contentWraper}>{content}</Box>
+        </Box>
+      </Dialog>
+      <Dialog
```
Comment (nrevutska) [inline src/components/popup-dialog/PopupDialog.tsx]: Why do we render two dialogs in this component?

**Review by nrevutska — CHANGES_REQUESTED** [review]

File: src/components/popup-dialog/PopupDialog.tsx (hunk 5 above)
Comment (geersann) [reply src/components/popup-dialog/PopupDialog.tsx]: Fixed it, add ConfirmDialog component

**Review by nrevutska — APPROVED** [review]

**Pull request merged**